sdk/servicebus/azure-servicebus/tests/async_tests/test_sb_client_async.py

```diff
@@ -443,3 +443,113 @@ async def test_backoff_fixed_retry(self):
         receiver._backoff(retried_times=1, last_exception=Exception('fake'), abs_timeout_time=None)
         sleep_time_fixed = time.time() - start_time
         assert sleep_time_fixed < backoff * (2 ** 1)
+
+    @pytest.mark.liveTest
+    @pytest.mark.live_test_only
+    @CachedResourceGroupPreparer(name_prefix='servicebustest')
+    @CachedServiceBusNamespacePreparer(name_prefix='servicebustest')
+    async def test_custom_client_id_queue_sender_async(self, servicebus_namespace_connection_string):
+        servicebus_connection_str = "connection_string"
+        queue_name = "queue_name"
+        custom_id = "my_custom_id"
+        servicebus_client = ServiceBusClient.from_connection_string(conn_str=servicebus_connection_str)
+        async with servicebus_client:
+            queue_sender = servicebus_client.get_queue_sender(queue_name=queue_name, client_identifier=custom_id)
+            assert queue_sender.identifier is not None
+            assert queue_sender.identifier == custom_id
```

Comment:
sorry I missed this before! I think you might need to add the queue preparer decorator, to check that the queue exists. I'm not sure how these tests are passing though, since the queue and connection string don't seem to exist?
```suggestion
    @CachedServiceBusNamespacePreparer(name_prefix='servicebustest')
    @CachedServiceBusQueuePreparer(name_prefix='servicebustest')
    async def test_custom_client_id_queue_sender_async(self, servicebus_namespace_connection_string, servicebus_queue):
        custom_id = "my_custom_id"
        servicebus_client = ServiceBusClient.from_connection_string(servicebus_namespace_connection_string)
        async with servicebus_client:
            queue_sender = servicebus_client.get_queue_sender(servicebus_queue.name, client_identifier=custom_id)
            assert queue_sender.identifier is not None
            assert queue_sender.identifier == custom_id
```
I believe you'll need to make this update to every queue test (adding the @CachedServiceBusQueuePreparer decorator, passing in the `servicebus_namespace_connection_string` from the args when creating the client, passing in the `servicebus_queue.name` from the args).